lib/blocks.php

```diff
@@ -102,7 +103,7 @@ function do_blocks( $content ) {
 	global $wp_registered_blocks;
 
 	// Extract the blocks from the post content.
-	$matcher = '/<!--\s*wp:([a-z](?:[a-z0-9\/]+)*)\s+((?:(?!-->).)*)\s*\/?-->(?:.*?<!--\s*\/wp:\g1\s+-->)?/';
+	$matcher = '/<!--\s*wp:([a-z](?:[a-z0-9\/]+)*)\s+((?:(?!-->).)*?)\s*\/?-->(?:.*?<!--\s*\/wp:\g1\s+-->)?/';
```

Comment:
we need to be extremely careful with combining wildcards. those `+)*` are troublesome, and I don't see why we are introducing `*?` - was it not working without the `?`? @westonruter